docs/glossary.md

```diff
@@ -19,6 +19,8 @@ title: "Glossary of Terms"
 
 [**Data Asset:**](./terms/data_asset.md) A collection of records within a Datasource which is usually named based on the underlying data system and sliced to correspond to a desired specification.
 
+[**Data Assistant:**](./terms/data_assistant.md) A utility that asks questions about your data, gathering information to describe what is observed, and then presents Metrics and proposes Expectations based on the answers.
```

Comment:
@Rachel-Reverie I would add to the definition something like "A goal-specific utility" (whereby goal could be Onboarding, Volume, etc.  Thanks!

Reply:
Updated in the next commit.  Thank you!